modules/library/src/main/scala/zio/elasticsearch/query/package.scala

```diff
@@ -18,31 +18,117 @@ package zio.elasticsearch
+     * @return
+     *   a new instance of the [[ElasticQuery]] with the `caseInsensitive` value set.
+     */
     def caseInsensitive(value: Boolean): Q
 
+    /**
+     * Sets the `caseInsensitive` parameter to `false` for this [[ElasticQuery]]. Same as [[caseInsensitive(false)]].
+     *
+     * @return
+     *   a new instance of the [[ElasticQuery]] with the `caseInsensitive` value set to `false`.
+     */
     final def caseInsensitiveFalse: Q = caseInsensitive(false)
 
+    /**
+     * Sets the `caseInsensitive` parameter to `true` for this [[ElasticQuery]]. Same as [[caseInsensitive(true)]].
+     *
+     * @return
+     *   a new instance of the [[ElasticQuery]] with the `caseInsensitive` value set to `true`.
+     */
     final def caseInsensitiveTrue: Q = caseInsensitive(true)
   }
 
   private[elasticsearch] trait HasIgnoreUnmapped[Q <: HasIgnoreUnmapped[Q]] {
+
+    /**
+     * Sets the `ignoreUnmapped` parameter to control whether to ignore unmapped fields and return empty hits
+     *
+     * @param value
+     *   the `boolean` value for `ignore_unmapped` parameter
+     * @return
+     *   a new instance of the [[ElasticQuery]] with the `ignoreUnmapped` value set.
+     */
     def ignoreUnmapped(value: Boolean): Q
 
+    /**
+     * Sets the `ignoreUnmapped` parameter to `false` for this [[ElasticQuery]]. Same as [[ignoreUnmapped(false)]].
+     *
+     * @return
+     *   a new instance of the [[ElasticQuery]] with the `ignoreUnmapped` value set to `false`.
+     */
     final def ignoreUnmappedFalse: Q = ignoreUnmapped(false)
 
+    /**
+     * Sets the `ignoreUnmapped` parameter to `true` for this [[ElasticQuery]]. Same as [[ignoreUnmapped(true)]].
+     *
+     * @return
+     *   a new instance of the [[ElasticQuery]] with the `ignoreUnmapped` value set to `true`.
+     */
     final def ignoreUnmappedTrue: Q = ignoreUnmapped(true)
   }
 
   private[elasticsearch] trait HasInnerHits[Q <: HasInnerHits[Q]] {
+
+    /**
+     * Sets the inner hits for this [[ElasticQuery]] to the default `InnerHits()` value.
+     *
+     * @return
+     *   a new instance of the [[ElasticQuery]] with the default inner hits.
+     */
     final def innerHits: Q = innerHits(InnerHits())
+
+    /**
+     * Sets the inner hits configuration for the [[NestedQuery]].
+     *
+     * @param innerHits
+     *   the configuration for inner hits
+     * @return
+     *   a new instance of the [[ElasticQuery]] with the specified inner hits configuration.
+     */
     def innerHits(innerHits: InnerHits): Q
   }
 
   private[elasticsearch] trait HasScoreMode[Q <: HasScoreMode[Q]] {
+
+    /**
+     * Sets the [[ScoreMode]] for the [[NestedQuery]]. The [[ScoreMode]] specifies how scores of matching documents are
+     * combined for the [[NestedQuery]].
+     *
+     * @param scoreMode
+     *   the [[ScoreMode]] to use for the [[NestedQuery]]
+     *   - [[ScoreMode.Avg]]: Use the mean relevance score of all matching child objects.
```

Comment:
`Uses` or `Use`? Maybe to avoid `.` and capital letters here.

"uses the mean relevance score of all matching child objects"